windows/check_quiesce_snapshot/check_quiescing_type.yml

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Initialize not support application consistent quiescing type"
+  ansible.builtin.set_fact:
+    winsrv_app_quiesce_support: false
+
+# Get *vss_manifests*.zip files in VM folder
+- name: "Get all VM files in VM folder"
+  include_tasks: ../../common/vm_get_config.yml
+  vars:
+    property_list:
+      - 'layoutEx.file'
+- name: "Failed to get VM files in VM folder"
+  ansible.builtin.assert:
+    that:
+      - vm_config.layoutEx.file is defined
+      - vm_config.layoutEx.file | length > 0
+    fail_msg: "Not get VM files info from 'vm_config.layoutEx.file' after taking snapshot."
+
+- name: "Set fact of the list of vss_manifests zip files"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_list_after: "{{ vm_config.layoutEx.file | map(attribute='name') | select('search', 'vss_manifests') }}"
+
+- name: "Display the vss_manifests zip files after taking quiesce snapshot"
+  ansible.builtin.debug: var=vm_vss_manifests_list_after
+
+- name: "Set fact of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_new: "{{ vm_vss_manifests_list_after | difference(vm_vss_manifests_list_before) }}"
+
+- name: "Check one new vss_manifests zip file generated"
+  ansible.builtin.assert:
+    that:
+      - vm_vss_manifests_new | length == 1
+    fail_msg: >
+      Before taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_before }},
+      after taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_after }},
+      not get the new generated zip file.
+
+- name: "Set fact of local path of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_path: "{{ current_test_log_folder }}/{{ vm_vss_manifests_new[0] | basename }}"
+    vm_vss_manifests_unzip_path: "{{ current_test_log_folder }}/{{ (vm_vss_manifests_new[0] | basename).split('.')[0] }}"
+
+- name: "Create local directory for unarchived files"
+  include_tasks: ../../common/create_directory.yml
+  vars:
+    dir_path: "{{ vm_vss_manifests_unzip_path }}"
+
+- name: "Get the new vss_manifests zip file to local"
+  include_tasks: ../../common/esxi_download_datastore_file.yml
+  vars:
+    src_datastore: "{{ datastore }}"
+    src_file_path: "{{ vm_vss_manifests_new[0].split(']')[1].strip() }}"
+    dest_file_path: "{{ vm_vss_manifests_path }}"
+
+- name: "Get the list of files in the new vss_manifests zip file"
+  ansible.builtin.unarchive:
+    src: "{{ vm_vss_manifests_path }}"
+    dest: "{{ vm_vss_manifests_unzip_path }}"
+    list_files: true
+  register: vm_vss_manifests_files
+
+# Windows Server 2025 VM with NVMe boot disk controller does not
+# suppport application consistent quiesce snapshot due to a known Windows issue.
+# Please keep track of the 3rd party issue status and vSphere support status.
+- name: "Get boot disk controller type in guest OS"
+  include_tasks: ../utils/win_get_boot_disk_ctl_type.yml
+  when: not new_vm
```

Comment:
`win_boot_disk_ctl_type` might be already set in vhba test cases. I think this task can have one more condition `win_boot_disk_ctl_type is undefined or not win_boot_disk_ctl_type`.

Even `win_boot_disk_ctl_type` can also be set in `deploy_vm` with the value of `boot_disk_controller`. Then below task can just check `win_boot_disk_ctl_type` no matter it is new vm or not.